azure-quantum/azure/quantum/target/target.py

```diff
@@ -31,8 +31,9 @@ def __init__(
         provider_id: str = "",
         content_type: ContentType = ContentType.json,
         encoding: str = "",
-        average_queue_time: float = None,
-        current_availability: str = ""
+        average_queue_time: Union[float, None] = None,
+        current_availability: str = "",
+        job_cls: Type[Job] = Job
```

Comment:
Can you add some comments to this class description around why these changes to Job/BaseJob are needed?

Reply:
Added documentation. Instead of adding it to `__init__` I can create a method to `Target` that the derived target can override.